plugins/flytekit-pydantic/flytekitplugins/pydantic/object_store.py

```diff
@@ -0,0 +1,107 @@
+import uuid
+from typing import Any, Dict, Type, TypeVar, cast
+
+import pandas as pd
+import torch.nn as nn
+from typing_extensions import Annotated, NewType
+
+from flytekit.core import context_manager, type_engine
+from flytekit.models import literals
+from flytekit.types import directory
+from flytekit.types.file import file
+
+PYDANTIC_SUPPORTED_FLYTE_TYPES = (
```

Comment:
Suggestion: What do you think about simply using [this](https://github.com/flyteorg/flytekit/blob/20427ee9cdbe068f17a3e1f67910fcd231a48c3d/flytekit/core/type_engine.py#L952) to retrieve the supported types from the type engine? Benefit would be that types that users create type transformers for would automatically be picked up.